DOCSUP-13742: Support PARTITION BY for the s3 table function (#28716)
ClickHouse/ClickHouse: docs/en/sql-reference/table-functions/s3.md

SELECT name, value FROM existing_table;
```

## Partitioned Write {#partitioned-write}

> If you specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data helps to improve reading operations efficiency.

Author: ```suggestion
If you specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data into separate files helps to improve reading operations efficiency.
```

> If you will specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data helps to improve reading operations efficiency.

Author: ```suggestion
If you specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data helps to improve reading operations efficiency.
```

> If you specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data into separate files helps to improve reading operations efficiency.

Reviewer: ```suggestion
If you specify `PARTITION BY` expression when inserting data into `S3` table, a separate objects is created for each partition value. Splitting the data into separate objects helps to improve reading operations efficiency.
```

> When inserting data into S3 table, you can split it into several files by specifying `PARTITION BY` expression. In this case a separate file is created for each partition value. Splitting the data helps to improve reading operations efficiency.

Reviewer 2: ```suggestion
If you will specify `PARTITION BY` expression when inserting data into `S3` table, a separate file is created for each partition value. Splitting the data helps to improve reading operations efficiency.
```

**Examples**

> 1. Using partition ID in a file name creates separate files:

Reviewer: ```suggestion
1. Using partition ID in a key creates separate objects:
```

> 1. Using partition ID in a file name creates separate files:

Reviewer: ```suggestion
1. Using partition ID in a key creates separate files:
```

```sql

> INSERT INTO TABLE FUNCTION
>     s3('http://bucket.amazonaws.com/my_bucket/file_{_partition_id}.csv', 'CSV', 'a UInt32, b UInt32, c UInt32')
>     PARTITION BY a VALUES (1, 2, 3), (1, 4, 5), (10, 11, 12), (10, 13, 14), (20, 21, 22), (20, 23, 24)

Author: ```suggestion
INSERT INTO TABLE FUNCTION
    s3('http://bucket.amazonaws.com/my_bucket/file_{_partition_id}.csv', 'CSV', 'a String, b UInt32, c UInt32')
    PARTITION BY a VALUES ('x', 2, 3), ('x', 4, 5), ('y', 11, 12), ('y', 13, 14), ('z', 21, 22), ('z', 23, 24)
```

```

> As a result, the data is written into three files: `file_1.csv`, `file_10.csv`, and `file_20.csv`.

Author: ```suggestion
As a result, the data is written into three files: `file_x.csv`, `file_y.csv`, and `file_z.csv`.
```

> As a result, the data is written into three files: `file_x.csv`, `file_y.csv`, and `file_z.csv`.

Reviewer: ```suggestion
As a result, the data is written into three objects: `file_x.csv`, `file_y.csv`, and `file_z.csv`.
```

> 2. Using partition ID in a bucket name creates files in different buckets:

Reviewer: ```suggestion
2. Using partition ID in a bucket name creates objects in different buckets:
```

```sql
INSERT INTO TABLE FUNCTION
    s3('http://bucket.amazonaws.com/my_bucket_{_partition_id}/file.csv', 'CSV', 'a UInt32, b UInt32, c UInt32')

>     PARTITION BY a VALUES (1, 2, 3), (1, 4, 5), (10, 11, 12), (10, 13, 14), (20, 21, 22), (20, 23, 24)

Author: ```suggestion
    PARTITION BY a VALUES (1, 2, 3), (1, 4, 5), (10, 11, 12), (10, 13, 14), (20, 21, 22), (20, 23, 24);
```

```

> As a result, the data is written into three files in different buckets: `my_bucket_1/file.csv`, `my_bucket_10/file.csv`, and `my_bucket_20/file.csv`.

Reviewer: ```suggestion
As a result, the data is written into three objects in different buckets: `my_bucket_1/file.csv`, `my_bucket_10/file.csv`, and `my_bucket_20/file.csv`.
```
